code/API/FreshdeskAPI.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7\Request;
+
+class FreshdeskAPI extends \Object
+{
+    /**
+     * This is used by the SilverStripe Injector to instantiate an object as $this->client. See _config/config.yml.
+     *
+     * @var array
+     */
+    private static $dependencies = [
+        'client' => '%$FreshdeskClient',
+    ];
+
+    /**
+     * Generic method to handle requests to the Freshdesk API.
+     *
+     * @param string $method, String $url, String $action, Array $headers, Array $data
+     * @param mixed  $url
+     * @param mixed  $action
+     * @param mixed  $data
+     * @param mixed  $headers
+     *
+     * @return $response || false
+     */
+    public function APICall($method, $url, $action, $data = [], $headers = ['Content-type' => 'application/json'])
```

Comment:
We already know it's an API from the class name. `call` should suffice!

Reply:
.. also known as "stuttering" `FreshdeskAPI::APICall()`

Reply:
ack